site/src/sphinx/server-annotated-service.rst

```diff
@@ -0,0 +1,718 @@
+
+Armeria also provides `@Header`_ annotation to inject an HTTP header value into a parameter. The parameter
+annotated with `@Header`_ is also specified as one of the built-in types as follows. `@Default`_ and
+``Optional<?>`` are also supported.
+
+.. code-block:: java
+
+    class MyAnnotatedServiceClass {
+
+        @Get("/hello1")
+        public HttpResponse hello1(@Header("Authorization") String auth) { ... }
+
+        @Post("/hello2")
+        public HttpResponse hello2(@Param("Content-Length") long contentLength) { ... }
+    }
+
+Moreover, the following classes are automatically injected into your parameter when you specify it on the
+parameter list of your method.
+
+- RequestContext_
+- ServiceRequestContext_
+- Request_
+- HttpRequest_
+- AggregatedHttpMessage_
+- HttpParameters_
+
+.. code-block:: java
+
+    class MyAnnotatedServiceClass {
+
+        @Get("/hello1")
+        public HttpResponse hello1(ServiceRequestContext ctx, HttpRequest req) {
+            // Use the context and request inside a method.
+        }
+
+        @Post("/hello2")
+        public HttpResponse hello2(AggregatedHttpMessage aggregatedMessage) {
+            // Armeria aggregates every message received, then call this method with an aggregated message.
+        }
+
+        @Get("/hello3")
+        public HttpResponse hello3(HttpParameters httpParameters) {
+            // 'httpParameters' holds parameters which are parsed from a query string of a request.
+        }
+
+        @Post("/hello4")
+        public HttpResponse hello4(HttpParameters httpParameters) {
+            // If a request has a url-encoded form as its body, it can be accessed via 'httpParameters'.
+        }
+    }
+
+Handling exceptions
+-------------------
+
+An exception may be raised from a service method, of course. Armeria provides `@ExceptionHandler`_
+annotation to transform an exception into a response. You can write your own exception handler
+by implementing ExceptionHandlerFunction_ interface and annotate your service object or method
+with `@ExceptionHandler`_ annotation. Here is an example of an exception handler. If your exception
+handler is not able to handle a given exception, you can call ``ExceptionHandlerFunction.fallthrough()``
+to pass the exception to the next exception handler.
+
+.. code-block:: java
+
+    class MyExceptionHandler implements ExceptionHandlerFunction {
+        @Override
+        public HttpResponse handleException(RequestContext ctx, HttpRequest req, Throwable cause) {
+            if (cause instanceof MyServiceException) {
+                return HttpResponse.of(HttpStatus.CONFLICT);
+            }
+
+            // To the next exception handler.
+            return ExceptionHandlerFunction.fallthrough();
+        }
+    }
+
+To catch an exception from every method in your service object, annotate your class with the annotation.
+
+.. code-block:: java
+
+    @ExceptionHandler(MyExceptionHandler.class)
+    class MyAnnotatedServiceClass {
+
```

Comment:
Probably OK to remove this empty line because this class has only a single method?